### PR TITLE
Delete organisation

### PR DESCRIPTION
**Pull request by KOTLAPRA23**
### JIRA link (if applicable) ###

RDCC-1760

### Change description ###

Delete organisation feature enabling in the prod.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/professionalapi/configuration/resolver/UserIdArgumentResolver.java [hunk 1]
```diff
@@ -35,6 +42,11 @@ public Object resolveArgument(
 
         HttpServletRequest request
                 = (HttpServletRequest) nativeWebRequest.getNativeRequest();
+
+        if (!deleteOrganisationEnabled && request.getMethod().equals("DELETE")) {
+
+            throw new AccessDeniedException(ERROR_MESSAGE_403_FORBIDDEN);
+        }
```
Comment (shashank-rastogi) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/configuration/resolver/UserIdArgumentResolver.java:49]: I dont think this is the right place to do the feature toggle. to keep things simple and easy to maintain, i would suggest you to add this at the controller level

File: src/main/java/uk/gov/hmcts/reform/professionalapi/configuration/FeatureConditionEvaluation.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    @Value("${deleteOrganisationEnabled}")
+    private boolean deleteOrganisationEnabled;
+
+    @Override
+    public boolean preHandle(HttpServletRequest request,
+                             HttpServletResponse response, Object arg2) throws Exception {
+
+        if (!deleteOrganisationEnabled && request.getMethod().equals("DELETE")) {
+
```
Comment (shashank-rastogi) [inline src/main/java/uk/gov/hmcts/reform/professionalapi/configuration/FeatureConditionEvaluation.java:24]: you can avoid the  if condition  if you do interceptor with API endpoint as you wont have to check for request method in that case

Comment (KOTLAPRA23) [reply src/main/java/uk/gov/hmcts/reform/professionalapi/configuration/FeatureConditionEvaluation.java:24]: ok

File: src/functionalTest/resources/application-functional.yaml [hunk 3]
```diff
@@ -1,4 +1,4 @@
-targetInstance: ${TEST_URL:http://localhost:8090}
+targetInstance: ${TEST_URL:http://rd-professional-api-pr-682.service.core-compute-preview.internal}
```
Comment (pksaxena01) [inline src/functionalTest/resources/application-functional.yaml:1]: Should be localhost

**Pull request merged**